pdf/creator/table.go

```diff
@@ -215,34 +216,61 @@ func (table *Table) GeneratePageBlocks(ctx DrawContext) ([]*Block, DrawContext,
 
 		if cell.backgroundColor != nil {
 			// Draw background (fill)
-			rect := NewRectangle(ctx.X, ctx.Y, w, h)
+			border := NewBorder(ctx.X, ctx.Y, w, h)
+			border.LineStyle = cell.borderLineStyle
 			r := cell.backgroundColor.R()
 			g := cell.backgroundColor.G()
 			b := cell.backgroundColor.B()
-			rect.SetFillColor(ColorRGBFromArithmetic(r, g, b))
+			border.SetFillColor(ColorRGBFromArithmetic(r, g, b))
 			if cell.borderStyle != CellBorderStyleNone {
 				// and border.
-				rect.SetBorderWidth(cell.borderWidth)
-				r := cell.borderColor.R()
-				g := cell.borderColor.G()
-				b := cell.borderColor.B()
-				rect.SetBorderColor(ColorRGBFromArithmetic(r, g, b))
+				if cell.borderColorLeft != nil {
+					border.SetColorLeft(ColorRGBFromArithmetic(cell.borderColorLeft.R(), cell.borderColorLeft.G(), cell.borderColorLeft.B()))
+				}
+				if cell.borderColorBottom != nil {
+					border.SetColorBottom(ColorRGBFromArithmetic(cell.borderColorBottom.R(), cell.borderColorBottom.G(), cell.borderColorBottom.B()))
+				}
+				if cell.borderColorRight != nil {
+					border.SetColorRight(ColorRGBFromArithmetic(cell.borderColorRight.R(), cell.borderColorRight.G(), cell.borderColorRight.B()))
+				}
+				if cell.borderColorTop != nil {
+					border.SetColorTop(ColorRGBFromArithmetic(cell.borderColorTop.R(), cell.borderColorTop.G(), cell.borderColorTop.B()))
+				}
+				border.SetWidthBottom(cell.borderWidthBottom)
+				border.SetWidthLeft(cell.borderWidthLeft)
+				border.SetWidthRight(cell.borderWidthRight)
+				border.SetWidthTop(cell.borderWidthTop)
 			} else {
```

Comment:
For double border, probably need a special case.
Typically looks like (Excel)
![image](https://user-images.githubusercontent.com/50777/42174626-898956c6-7e12-11e8-8a0a-5e3213bb46fb.png)
where the middle cell has a double bottom border, and the ones on left and right have a regular solid bottom border.